extensions/kogito/deployment/src/main/java/io/quarkus/kogito/deployment/KogitoAssetsProcessor.java

```diff
@@ -150,10 +247,12 @@ private ApplicationGenerator createApplicationGenerator(Path projectPath, Launch
             throws IOException {
 
         Path srcPath = projectPath.resolve("src");
-        String appPackageName = "org.kie.kogito.app";
+        boolean usePersistence = combinedIndexBuildItem.getIndex()
+                .getClassByName(createDotName("org.kie.kogito.persistence.KogitoProcessInstancesFactory")) != null;
```

Comment:
You may want to create a constant for this.

Reply:
updated